typify-impl/src/type_entry.rs

```diff
@@ -1635,7 +1636,10 @@ impl TypeEntry {
                 } else {
                     let key_ident = key_ty.type_ident(type_space, type_mod);
                     let value_ident = value_ty.type_ident(type_space, type_mod);
-                    quote! { ::std::collections::HashMap<#key_ident, #value_ident> }
+
+                    let map_to_use = syn::parse_str::<syn::TypePath>(map_to_use)
```

Comment:
can we parse this much earlier? e.g. in the call to the settings object? It would be annoying to fail so far from the source

Reply:
We can, but storing it as a `syn::Type` will complicate the rest of the code a fair bit. We'll have to do a `.to_token_stream().into_string()` for the deserialize call, and it doesn't implement `Debug` which means we'd need to switch to a custom `Debug` implementation on the `TypeSpaceSettings` object.

Reply:
I can try to parse it immediately, but still store it as a String - but then we are doing double the work to try and parse it twice. :P

Let me know how you'd like me to proceed!

Reply:
Right now we're parsing it for every map. Can we store it as a TokenStream?

Reply:
Similar complexity gain with either `syn::Type` or `TokenStream`. I see the value in avoiding the extra parsing and so have added the code in to store this as a `syn::Type`. :)